### ISSUE TITLE
[Feature Request] Add "one node per shard" routing rule to support distributed query processing

### ISSUE THREAD

**Issue by joehu888** [redis-rs/redis-rs#734]
Current request routing supports AllNodes/AllMasters routing. What's missing is "one node per shard" routing. The use case is distributed transaction processing. e.g., I have a module command called M.QUERY, which performs per-shard query. I'd do Map/Reduce like distributed query:
1. Route the command M.QUERY to "one node per shard" (ie., either master or replica node of each shard).
2. Aggregate results from each shard and send the consolidated result back to the client.

Using the current AllNodes/AllMasters routing, the command is only executed on all master. Replicas are not utilized for query, which makes it not scale well. (BTW, it seems to me that AllNodes is same as AllMasters. Can someone confirm? I thought AllNodes is supposed to include all replicas?)

So, I'm proposing to add a "one node per shard" routing. Does the current code base have the infrastructure to support routing a ReadOnly command to a replica? How much work will be involved to implement the feature?

**Comment by doyshinda** [comment]
> Does the current code base have the infrastructure to support routing a ReadOnly command to a replica?

Yes, use the [read_from_replicas](https://docs.rs/redis/latest/redis/cluster/struct.ClusterClientBuilder.html#method.read_from_replicas) flag when building a connection.

With this flag set, `AllNodes` becomes "1 replica from each shard" (see code [here](https://github.com/redis-rs/redis-rs/blob/main/redis/src/cluster.rs#L224) ).

Once #732 is implemented, I believe this feature could be accomplished by creating the connection with `read_from_replicas` and  the setting the custom routing to `RoutingInfo::AllNodes`.
